solana/src/main/java/com/solana/networking/RpcResponse.kt

```diff
@@ -15,15 +15,18 @@ typealias DefaultRpcResponse = RpcResponse<JsonElement>
 @Serializable
 data class RpcError(val code: Int, val message: String)
 
+interface SolanaResponse<R> {
```

Comment:
So when I originally created the RpcResponse object, I based it purely off of the [JSON-RPC 2.0 Specification](https://www.jsonrpc.org/specification) in order to separate the RPC stuff from Solana stuff. 

So IMO, the naming here should reflect that. The base class should be JsonRpcResponse or something similar. 

I also think this should be a base class rather than an interface. In such a simple case its really not a big difference either way (interface vs base class), but I think a base class makes more sense. Interfaces should define _behavior_ whereas classes define _structure_. Since this JSON-RPC 2.0 response is an expected data _structure_ that we receive, and the Socket Response is an _extension_ of that response, a class inheritance topography makes the most sense here.

Reply:
just my opinion tho, curious to hear your thoughts

Reply:
I agree